packages-next/fields/src/types/integer/views/index.tsx

```diff
@@ -15,15 +15,18 @@ import {
 export const Field = ({ field, value, onChange, autoFocus }: FieldProps<typeof controller>) => (
   <FieldContainer>
     <FieldLabel>{field.label}</FieldLabel>
-    <TextInput
-      autoFocus={autoFocus}
-      type="number"
-      readOnly={onChange === undefined}
-      onChange={event => {
-        onChange?.(event.target.value.replace(/\D/g, ''));
-      }}
-      value={value}
-    />
+    {onChange ? (
+      <TextInput
+        autoFocus={autoFocus}
+        type="number"
+        onChange={event => {
+          onChange(event.target.value.replace(/\D/g, ''));
+        }}
+        value={value}
+      />
+    ) : (
+      value
```

Comment:
Does this need to be in a div?

Reply:
Not at the moment, it's consistent with other fields for now.

Ultimately I do think we should add a wrapper of some sort to readonly output (including the Virtual field) but am going to leave that for a separate design session to work out.